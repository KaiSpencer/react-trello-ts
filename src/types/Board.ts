import {CSSProperties} from 'react'

export interface BoardData {
  lanes: Lane[]
}

export interface Lane {
  id: string
  title?: string
  label?: string
  style?: CSSProperties
  cards: Card[]
  currentPage?: number
  droppable?: boolean
  labelStyle?: CSSProperties
  cardStyle?: CSSProperties
  disallowAddingCard?: boolean
  [key: string]: any
}

export interface Card {
  id: string
  title?: string
  label?: string
  description?: string
  laneId?: string
  style?: CSSProperties
  draggable?: boolean
  [key: string]: any
}
