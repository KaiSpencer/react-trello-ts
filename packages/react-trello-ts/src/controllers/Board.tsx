import React, { createContext, FC, PropsWithChildren } from "react";
import classNames from "classnames";
import uuidv1 from "uuid/v1";
import { BoardContainer } from "./BoardContainer";
import * as DefaultComponets from "../components";
import { BoardData } from "../types/Board";
import { store } from "../store/store";

export const Board: FC<
	PropsWithChildren<{
		id?: string;
		className?: string;
		components: typeof DefaultComponets;
		data: BoardData;
		t?: any;
	}>
> = ({ data, children, className, components, id, t, ...rest }) => {
	const allClassNames = classNames("react-trello-board", className || "");

	return (
		<BoardContext.Provider value={store}>
			<components.GlobalStyle />
			<BoardContainer
				id={id || uuidv1()}
				data={data}
				components={components}
				t={t}
				className={allClassNames}
				{...rest}
			/>
		</BoardContext.Provider>
	);
};

export const BoardContext = createContext(store);
