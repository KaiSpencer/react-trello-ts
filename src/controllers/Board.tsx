import React, {createContext, FC, PropsWithChildren} from 'react'
import classNames from 'classnames'
import uuidv1 from 'uuid/v1'
import {BoardContainer} from './BoardContainer'
import * as DefaultComponets from '../components'
import {BoardData} from 'rt/types/Board'
import {store} from 'rt/store/store'

// export class _Board extends Component<{
//   id?: string
//   className?: string
//   components?: typeof DefaultComponets
//   data: BoardData
//   t?: any
// }> {
//   store: any
//   id: any
//   constructor(props) {
//     super(props)
//     this.store = this.getStore()
//     this.id = this.props.id || uuidv1()
//   }

//   getStore = () => {
//     //When you create multiple boards, unique stores are created for isolation
//     return createStore(boardReducer)
//   }

//   render() {
//     const {className, components} = this.props
//     const allClassNames = classNames('react-trello-board', className || '')
//     return (
//       <Provider store={this.store}>
//         <>
//           <components.GlobalStyle />
//           <BoardContainer id={this.id} {...this.props} className={allClassNames} />
//         </>
//       </Provider>
//     )
//   }
// }

export const Board: FC<
  PropsWithChildren<{
    id?: string
    className?: string
    components: typeof DefaultComponets
    data: BoardData
    t?: any
  }>
> = ({data, children, className, components, id, t, ...rest}) => {
  const allClassNames = classNames('react-trello-board', className || '')

  return (
    <BoardContext.Provider value={store}>
      <components.GlobalStyle />
      <BoardContainer
        id={id || uuidv1()}
        data={data}
        components={components}
        t={t}
        className={allClassNames}
        {...rest}
      />
    </BoardContext.Provider>
  )
}

export const BoardContext = createContext(store)
