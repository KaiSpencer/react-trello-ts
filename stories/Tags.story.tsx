import { Meta, StoryObj, storiesOf } from "@storybook/react";
import React from "react";

import Board from "../src";
Board.displayName = "Board";

export default ({
	title: "Basic Functions",
	component: Board,
} satisfies Meta<typeof Board>);

type Story = StoryObj<typeof Board>;

export const Tags: Story = {
	name: "Tags",
	render: (args) => {
		const data = {
			lanes: [
				{
					id: "lane1",
					title: "Planned Tasks",
					cards: [
						{
							id: "Card1",
							title: "Card1",
							description: "foo card",
							metadata: { cardId: "Card1" },
							tags: [
								{ title: "High", color: "white", bgcolor: "#EB5A46" },
								{ title: "Tech Debt", color: "white", bgcolor: "#0079BF" },
								{
									title: "Very long tag that is",
									color: "white",
									bgcolor: "#61BD4F",
								},
								{ title: "One more", color: "white", bgcolor: "#61BD4F" },
							],
						},
						{
							id: "Card2",
							title: "Card2",
							description: "bar card",
							metadata: { cardId: "Card2" },
							tags: [{ title: "Low" }],
						},
					],
				},
			],
		};
		return <Board data={data} tagStyle={{ fontSize: "80%" }} />;
	},
};
