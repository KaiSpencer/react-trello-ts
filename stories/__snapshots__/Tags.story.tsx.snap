// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Functions Tags smoke-test 1`] = `
<div draggable="false"
     class="sc-iGgWBj iSNfA"
>
  <div>
    <div class="smooth-dnd-container horizontal">
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Planned Tasks
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Card1"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Card1
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  foo card
                </div>
                <div class="sc-eldPxv iDFVov">
                  <span class="sc-fPXMVe eCIUDc"
                        style="color: white; background-color: rgb(235, 90, 70); font-size: 80%;"
                  >
                    High
                  </span>
                  <span class="sc-fPXMVe eCIUDc"
                        style="color: white; background-color: rgb(0, 121, 191); font-size: 80%;"
                  >
                    Tech Debt
                  </span>
                  <span class="sc-fPXMVe eCIUDc"
                        style="color: white; background-color: rgb(97, 189, 79); font-size: 80%;"
                  >
                    Very long tag that is
                  </span>
                  <span class="sc-fPXMVe eCIUDc"
                        style="color: white; background-color: rgb(97, 189, 79); font-size: 80%;"
                  >
                    One more
                  </span>
                </div>
              </article>
            </div>
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Card2"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Card2
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  bar card
                </div>
                <div class="sc-eldPxv iDFVov">
                  <span class="sc-fPXMVe eCIUDc"
                        style="color: white; background-color: orange; font-size: 80%;"
                  >
                    Low
                  </span>
                </div>
              </article>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;
